### DISCUSSION TITLE
Why ORA-06413 but database/sql/driver.ErrBadConn

### DISCUSSION THREAD

**Discussion by KacperPerschke** [sijms/go-ora#473] [Q&A] [answered]
My microservice using `go-ora/v2` encountered a situation where it lost connections to the database, which was turned off for a long time due to maintenance reasons. It was unable to re-establish the connection and permanently returned error `ORA-06413`.

From static code analysis, I see that in some situations you are intentionally not qualifying `ORA-06413` as `database/sql/driver.ErrBadConn` which would allow `sql` to attempt to reconnect. Please provide the reasons for this approach.

I need to find a solution to the problem described at the beginning. _I want to do it according to your design decisions_.

**Comment by sijms** [comment]
sorry for late
I will update the code and inform you

**Comment by sijms** [comment, marked as answer]
I update the code to return `driver.ErrBadConn` instead of ora-06413 when connection is not valid
it will be available in next release

**Reply by KacperPerschke** [reply]
Thanks!
